File: 3DTools/LightSource.cpp
#include "LightSource.h"
namespace DDD {
	LightSource::LightSource(std::string serialized) :Vanta(cgtools::Color(0, 0, 0)) {
		load(serialized);
	}
	LightSource::LightSource(cgtools::Color Emission, double overdrive)noexcept : Vanta(Emission* overdrive) {}
	std::string LightSource::serialize() const {
		return AMaterial::includeClassID(emission.serialize(), CLASSID);
	}
	void LightSource::load(std::string serialized) {
		emission.load(serialized);
	}
	AMaterial* LightSource::clone() const {
		return new LightSource(emission, 1);
	}
}

File: RayTracerInterface/InterfacingClasses/Material/LightSource.cs
namespace RayTracerInterface {
	public class LightSource : AMaterial {
		public const int TID = 3;
		public override string Serialize() => new Color(Emission.r * Overdrive, Emission.g * Overdrive, Emission.b * Overdrive).Serialize();
		public override int TYPEID() => TID;
		public double Overdrive = 1;
	}
}